### PR TITLE
Story 5

### PR DESCRIPTION
**Pull request by michael-bertram**
Created the return a book form on single Book Detail component

### CODE REVIEW

**Comment by shellycs50** [comment]
Any issues here will be fixed in later merge as 'future' branches have 'better' implementations

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { useState } from "react"
+
+function BookReturnForm ({id, bookclaim}) {
+    const [email, setEmail] = useState ('')
+
+    function submit(event){
+        console.log('Email state variable: ', email, '     event.... value:', event.target.email.value)
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: Can this be removed, please?

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        event.preventDefault()
+        console.log('submitted returned book (attempt)!')
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: And this one.

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { useState } from "react"
+
+function BookReturnForm ({id, bookclaim}) {
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: camelCase for variables, please.

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    function submit(event){
+        console.log('Email state variable: ', email, '     event.... value:', event.target.email.value)
+        event.preventDefault()
+        console.log('submitted returned book (attempt)!')
+
+        fetch('https://book-swap-api.dev.io-academy.uk/api/books/return/' + id, {
+            method: "PUT",
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json",
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: The `Accept` header is missing.

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            },
+            body: JSON.stringify({
+                email: event.target.email.value,
+            }),
+        }) 
+        .then((res) => {
+            if (res.ok) {
+                console.log('Return request submitted successfully.');
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: Remove the console.log, please?

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                bookclaim(null)
+                // Successfully Claimed
+            } else {
+                console.error('Failed to submit Return request.');
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: console.log again, please.

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                // Error in Claiming
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: There error is not being output here as outlined in the requirements.

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            }
+        })
+        .catch((error) => {
+            console.error('Error:', error);
+            // Some other error. issue apology.
+            console.log('something else went wrong with Returning')
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: Remove console log

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        console.log('submitted returned book (attempt)!')
+
+        fetch('https://book-swap-api.dev.io-academy.uk/api/books/return/' + id, {
+            method: "PUT",
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+                email: event.target.email.value,
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: This is not using state.

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    }
+//Do we also need to input function where if successful change claimed=0??
+//  To make the book change from claimed to available?
+
+    return (
+        <div>
+            <h2>Would you like to return this book?</h2>
+            <form onSubmit={submit}>
+                <label htmlFor="Email" value={email}>Email</label>
+                <input type="email" id='email' onChange={setEmail}/>
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: The onChange here is not working as expected.

File: src/components/BookReturnForm/BookReturnForm.jsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+    }
+//Do we also need to input function where if successful change claimed=0??
+//  To make the book change from claimed to available?
+
+    return (
+        <div>
+            <h2>Would you like to return this book?</h2>
+            <form onSubmit={submit}>
+                <label htmlFor="Email" value={email}>Email</label>
```
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: There is no value attribute on a HTML label.

File: src/components/Booklisting/SingleBookDetail/SingleBookDetail.jsx [hunk 12]
```diff
@@ -46,24 +47,21 @@ function SingleBookDetail () {
...
+              <h1>{title}</h1>
+              <p><strong>Author:</strong> {author}</p>
+              <p><strong>Published:</strong> {year}</p>
+              <p><strong>Pages:</strong> {pageCount}</p>
+              <p><strong>Genre:</strong> {genre}</p>
+              <p><strong>Blurb:</strong></p>
+              <p className="blurb">{blurb}</p> 
+            </div>
+        </div>
+        <div className="display_container form">
```
Comment (Geoff10) [inline src/components/Booklisting/SingleBookDetail/SingleBookDetail.jsx]: This component has 2 root elements and they are only allowed one, this means the site is not loaading at the moment.

File: src/components/BookReturnForm/BookReturnForm.jsx (hunk 11 above)
Comment (Geoff10) [inline src/components/BookReturnForm/BookReturnForm.jsx]: For is not working, I think it's the capital letter.

**Review by Geoff10 — CHANGES_REQUESTED** [review]
There are a few things not quite right, I think this may have been caused by merge conflicts.

**Review by Geoff10 — APPROVED** [review]
Nice work!

**Pull request merged**